### PR TITLE
Add helper function to get the fit_time and gen_time in each trial (not just aggregate)

### PR DESCRIPTION
**Pull request by esantorella**
Summary: This will be useful for benchmarking

Differential Revision: D44434115

### CODE REVIEW

File: ax/core/tests/test_utils.py [hunk 1]
```diff
@@ -185,3 +189,14 @@ def test_feasible_hypervolume(self) -> None:
...
         self.assertEqual(list(feas_hv), [0.0, 0.0, 1.0, 1.0])
+
+    def test_get_model_times(self) -> None:
+        exp = get_robust_branin_experiment(num_sobol_trials=2)
+        fit_times, gen_times = get_model_trace_of_times(exp)
+        total_fit_time, total_gen_time = get_model_times(exp)
+        fit_times_not_none = [none_throws(elt) for elt in fit_times]
+        gen_times_not_none = [none_throws(elt) for elt in gen_times]
+        self.assertTrue([elt > 0 for elt in fit_times_not_none])
+        self.assertTrue([elt > 0 for elt in gen_times_not_none])
```
Comment (Balandat) [inline ax/core/tests/test_utils.py:200]: This only asserts that the list is non-empty (i.e. this passes even if every list element is `False`) - is this what you want? Or do you want to assert that `all(elt > 0 for elt in fit_times_not_none)` is `True`?

File: ax/core/utils.py [hunk 2]
```diff
@@ -148,21 +150,37 @@ def best_feasible_objective(  # pragma: no cover
     return accumulate(f)
 
 
+def _extract_generator_run(trial: BaseTrial) -> GeneratorRun:
+    if isinstance(trial, BatchTrial):
+        return trial._generator_run_structs[0].generator_run
```
Comment (Balandat) [inline ax/core/utils.py:155]: This assumes the trial was generated by one generator run (in some cases there may be more, like in the online experimentation setting). We should probably raise an informative `UnsupportedError` or `NotImplementedError` here if that's not the case (and we don't want to implement the general case for now).

File: ax/core/utils.py [hunk 3]
```diff
@@ -148,21 +150,37 @@ def best_feasible_objective(  # pragma: no cover
...
-            gr = not_none(trial.generator_run)
-        else:
-            raise ValueError("Unexpected trial type")  # pragma: no cover
-        if gr.fit_time is not None:
-            fit_time += not_none(gr.fit_time)
-        if gr.gen_time is not None:
-            gen_time += not_none(gr.gen_time)
+    fit_times, gen_times = get_model_trace_of_times(experiment)
+    fit_time = sum([t for t in fit_times if t is not None])
+    gen_time = sum([t for t in gen_times if t is not None])
```
Comment (Balandat) [inline ax/core/utils.py:185]: Can sum directly on a generator
```suggestion
    fit_time = sum(t for t in fit_times if t is not None)
    gen_time = sum(t for t in gen_times if t is not None)
```

File: ax/core/tests/test_utils.py (hunk 1 above)
Comment (esantorella) [reply ax/core/tests/test_utils.py:200]: Oops, yes, I wanted the latter!

**Pull request closed without merging**